src/js/menu/menu-button.js

```diff
@@ -281,24 +282,26 @@ class MenuButton extends Component {
    * @listens keydown
    */
   handleKeyPress(event) {
-
-    // Escape (27) key or Tab (9) key unpress the 'button'
-    if (event.which === 27 || event.which === 9) {
+    // Escape or Tab unpress the 'button'
+    if (keycode.isEventKey(event, 'Esc') || keycode.isEventKey(event, 'Tab')) {
       if (this.buttonPressed_) {
         this.unpressButton();
       }
       // Don't preventDefault for Tab key - we still want to lose focus
-      if (event.which !== 9) {
+      if (!keycode.isEventKey(event, 'Tab')) {
         event.preventDefault();
         // Set focus back to the menu button's button
-        this.menuButton_.el_.focus();
+        this.menuButton_.focus();
       }
-    // Enter (13) or Up (38) key or Down (40) key press the 'button'
-    } else if (event.which === 13 || event.which === 38 || event.which === 40) {
+    // Up Arrow or Down Arrow also 'press' the button to open the menu
+    } else if (keycode.isEventKey(event, 'Up') || keycode.isEventKey(event, 'Down')) {
       if (!this.buttonPressed_) {
-        this.pressButton();
         event.preventDefault();
+        this.pressButton();
       }
+    } else {
+      // NOTE: This is a special case where we don't pass unhandled
+      //  keypress events up to the Component handler
```

Comment:
Would be good to add a reason in this comment, as well and the one below it in `handleSubmenuKeyPress`